lib/active_record_shards/connection_switcher.rb

```diff
@@ -111,10 +110,19 @@ def on_cx_switch_block(which, force: false, construct_ro_scope: nil, &block)
         readonly.scoping(&block)
       end
     ensure
-      @disallow_replica -= 1 if which == :primary
+      self.disallow_replica -= 1 if which == :primary
       switch_connection(old_options) if old_options
     end
 
+    def disallow_replica=(value)
+      @disallow_replica_by_thread[Thread.current] = value
```

Comment:
Why `@disallow_replica_by_thread[Thread.current]` and not e.g. `Thread.current[:disallow_replica_by_thread]`?